tests/ClientBuilderTest.php

```diff
@@ -221,4 +221,30 @@ public function testCompatibilityHeaderDefaultIsOff()
             $this->assertNotContains('application/vnd.elasticsearch+json;compatible-with=7', $request['request']['headers']['Accept']);
         }
     }
+
+    public function testFromConfigWithIncludePortInHostHeader()
+    {
+        $url = 'localhost:1234';
+        $config = [
+            'hosts' => [$url],
+            'includePortInHostHeader' => true,
+            'connectionParams' => [
+                'client' => [
+                    'verbose' => true
+                ]
+            ],
+        ];
+
+        $client = ClientBuilder::fromConfig($config);
+
+        $this->assertInstanceOf(Client::class, $client);
+
+        try {
+            $client->info();
+        } catch (OpenSearchException $e) {
+            $request = $client->transport->getLastConnection()->getLastRequestInfo();
+            $this->assertTrue(isset($request['request']['headers']['Host'][0]));
+            $this->assertEquals($url, $request['request']['headers']['Host'][0]);
+        }
```

Comment:
Is this test supposed to throw? If it does, and if `$client -> info()` doesn't throw, the test passes, which would be incorrect. You should assert failure after that call.

Reply:
I added the exception check